test/functional/feature_block.py

```diff
@@ -725,7 +731,6 @@ def run_test(self):
         assert(len(out[17].vout) < 42)
         tx.vin.append(CTxIn(COutPoint(out[17].sha256, 42), CScript([OP_TRUE]), 0xffffffff))
         tx.vout.append(CTxOut(0, b""))
-        tx.calc_sha256()
```

Comment:
At all these places, `tx.calc_256()` call was before `update_block()` which forcefully rehash the cached value.